api/prometheus/v1/api_refactored_test.go

```diff
@@ -0,0 +1,501 @@
+	`)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer suite.Close()
+	if err := suite.Run(); err != nil {
+		t.Fatal(err)
+	}
+	rr := testRulesRetriever{
+		testing: t,
+	}
+	db := func() apiv1.TSDBAdmin {
+		return &tsdb.DB{}
+	}
+	logger := log.NewNopLogger()
+
+	// create v1 api
+	api := apiv1.NewAPI(
+		suite.QueryEngine(),
+		suite.Storage(),
+		testTargetRetriever{},
+		testAlertmanagerRetriever{},
+		func() config.Config { return samplePrometheusCfg },
+		sampleFlagMap,
+		func(f http.HandlerFunc) http.HandlerFunc { return f },
+		db,
+		true,
+		logger,
+		rr,
+		0,
+		0,
+		regexp.MustCompile(".*"),
+	)
+
+	// register the router with the api
+	router := route.New()
+	api.Register(router)
+	mux.Handle("/", http.StripPrefix(apiPrefix, router))
+
+	// do methods for API interfaces
+	doAlertManagers := func() func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.AlertManagers(context.Background())
+		}
+	}
+
+	doQuery := func(q string, ts time.Time) func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.Query(context.Background(), q, ts)
+		}
+	}
+
+	doQueryRange := func(q string, rng Range) func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.QueryRange(context.Background(), q, rng)
+		}
+	}
+
+	doLabelValues := func(label string) func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.LabelValues(context.Background(), label)
+		}
+	}
+
+	doSeries := func(matcher []string, startTime time.Time, endTime time.Time) func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.Series(context.Background(), matcher, startTime, endTime)
+		}
+	}
+
+	doRules := func() func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.Rules(context.Background())
+		}
+	}
+
+	doTargets := func() func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.Targets(context.Background())
+		}
+	}
+
+	doConfig := func() func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.Config(context.Background())
+		}
+	}
+
+	doFlags := func() func() (interface{}, error) {
+		return func() (interface{}, error) {
+			return promAPI.Flags(context.Background())
+		}
+	}
+
+	//doCleanTombstones := func() func() (interface{}, error) {
+	//	return func() (interface{}, error) {
+	//		return nil, promAPI.CleanTombstones(context.Background())
+	//	}
+	//}
+	//doDeleteSeries := func(matcher string, startTime time.Time, endTime time.Time) func() (interface{}, error) {
+	//	return func() (interface{}, error) {
+	//		return nil, promAPI.DeleteSeries(context.Background(), []string{matcher}, startTime, endTime)
+	//	}
+	//}
+	//doSnapshot := func(skipHead bool) func() (interface{}, error) {
+	//	return func() (interface{}, error) {
+	//		return promAPI.Snapshot(context.Background(), skipHead)
+	//	}
+	//}
+
```

Comment:
The commented functions call admin api endpoints. Should I use a fakeDB for these, as in here:

https://github.com/prometheus/prometheus/blob/7f7b21104734fd1c7a5d2b3bc812da96cd7c35a3/web/api/v1/api_test.go#L1006-L1020

Reply:
yes I think this will do.